ps3.py

```diff
@@ -0,0 +1,276 @@
+    num_vowels = int(math.ceil(n / 3))
+
+    for i in range(num_vowels):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1
+
+    for i in range(num_vowels, n):
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+    hand["*"] = 1
+    for letter in hand:
+        if letter in VOWELS and hand[letter] == 1:
+            del hand[letter]
+            return hand
+
+
+def update_hand(hand, word):
+    """
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+    """
+    new_hand = hand.copy()
+    for letter in word.lower():
+        if letter in new_hand and new_hand[letter] > 0:
+            new_hand[letter] -= 1
+    return new_hand
+
+
+def is_valid_word(word, hand, word_list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+    """
+    word = word.lower()
+    if "*" in word:
+        word = word.replace("*", f"[{VOWELS}]")
+        for word_g in word_list:
+            if re.fullmatch(word, word_g) is not None:
+                return True
+        else:
+            return False
+    elif word in word_list:
+        for letter in word:
+            if letter not in hand or word.count(letter) > hand[letter]:
+                return False
+        else:
+            return True
+    else:
+        return False
+
+
+
+
+
+def calculate_handlen(hand):
+    """
+    Returns the length (number of letters) in the current hand.
+    """
+    count_letter = 0
+    for letter in hand.keys():
+        count_letter += hand[letter]
+    return count_letter
+
+
+def play_hand(hand, word_list, count_points):
+    """
+    Allows the user to play the given hand, as follows:
+
+    * The hand is displayed.
+
+    * The user may input a word.
+
+    * When any word is entered (valid or invalid), it uses up letters
+      from the hand.
+
+    * An invalid word is rejected, and a message is displayed asking
+      the user to choose another word.
+
+    * After every valid word: the score for that word is displayed,
+      the remaining letters in the hand are displayed, and the user
+      is asked to input another word.
+
+    * The sum of the word scores is displayed when the hand finishes.
+
+    * The hand finishes when there are no more unused letters.
+      The user can also finish playing the hand by inputing two
+      exclamation points (the string '!!') instead of a word.
+    """
+    while calculate_handlen(hand) != 0:
+        print("Current Hand:", end=" ")
+        display_hand(hand)
+        user_word = input("Enter word, or “!!” to indicate that you are finished:")
+        if is_valid_word(user_word, hand, word_list):
+            count_points += get_word_score(user_word, calculate_handlen(hand))
+            print(f"{user_word} earned {str(get_word_score(user_word, calculate_handlen(hand)))} ."
```

Comment:
These instructions are repeating. It will be more simple to decrale new var with `get_word_score` value and pass it here.